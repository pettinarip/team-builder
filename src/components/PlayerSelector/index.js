import PlayerSelector from './PlayerSelector'

export default PlayerSelector
