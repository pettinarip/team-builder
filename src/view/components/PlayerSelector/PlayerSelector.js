import React from 'react'
import PropTypes from 'prop-types'
import PlayerDraggable from '../PlayerDraggable'
import AddPlayerButton from '../AddPlayerButton'

const PlayerSelector = ({ players, ...props }) => (
  <div className='list'>
    <AddPlayerButton onClick={props.onAddPlayerClick} />
    {players.map(player =>
      <PlayerDraggable
        className='item active'
        id={player.id}
        color={player.color}
        number={player.number}
        onDropped={props.onAddPlayerPosition}
        onDroppedOutside={props.onRemovePlayerPosition}
        key={player.id}
      />
    )}
  </div>
)

PlayerSelector.propTypes = {
  players: PropTypes.array.isRequired,
  onAddPlayerClick: PropTypes.func.isRequired,
  onAddPlayerPosition: PropTypes.func.isRequired
}

export default PlayerSelector
