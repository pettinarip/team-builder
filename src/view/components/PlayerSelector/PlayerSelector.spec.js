import React from 'react'
import { shallow } from 'enzyme'
import PlayerSelector from './PlayerSelector'

describe('PlayerSelector component', () => {
  it('should render an add player button', () => {
    const component = mountPlayerSelector()
    expect(component).toMatchSnapshot()
  })

  it('should render all players', () => {
    const players = [{ id: 1 }, { id: 2 }, { id: 3 }]
    const component = mountPlayerSelector({
      players
    })

    expect(component).toMatchSnapshot()
  })
})

function mountPlayerSelector (props) {
  const propsToUse = {
    onAddPlayerClick () {},
    onAddPlayerPosition () {},
    onRemovePlayerPosition () {},
    players: [],
    ...props
  }

  return shallow(<PlayerSelector {...propsToUse} />)
}
