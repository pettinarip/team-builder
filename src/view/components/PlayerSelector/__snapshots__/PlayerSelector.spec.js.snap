// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PlayerSelector component should render all players 1`] = `
<div
  className="list"
>
  <AddPlayerButton
    onClick={[Function]}
  />
  <DragSource(PlayerDraggable)
    className="item active"
    id={1}
    onDropped={[Function]}
    onDroppedOutside={[Function]}
  />
  <DragSource(PlayerDraggable)
    className="item active"
    id={2}
    onDropped={[Function]}
    onDroppedOutside={[Function]}
  />
  <DragSource(PlayerDraggable)
    className="item active"
    id={3}
    onDropped={[Function]}
    onDroppedOutside={[Function]}
  />
</div>
`;

exports[`PlayerSelector component should render an add player button 1`] = `
<div
  className="list"
>
  <AddPlayerButton
    onClick={[Function]}
  />
</div>
`;
